platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,826 @@
+anyone, the consensus nodes must be able to aggregate their individual signatures on a message into a valid ledger
+signature for the message that is verifiable by the ledger id.
+
+The TSS effort has been broken down into four separate proposals: TSS-Library, TSS-Roster, TSS-Ledger-ID, and
+TSS-Block-Signing.
+
+1. The `TSS-Library` proposal contains the cryptographic primitives and algorithms needed to implement TSS.
+2. The `TSS-Roster` proposal introduces the data structure of a consensus `Roster` to replace the platform's concept of
+   an `AddressBook` and modifies the life-cycle for when the platform receives new consensus rosters.
+3. This proposal (`TSS-Ledger-ID`) depends on the first two proposals and is for the integration of a threshold
+   signature scheme into the consensus node, delivering the ability for the ledger to sign a message with the
+   ledger private key.
+4. The `TSS-Block-Signing` proposal is everything needed to support the signing of blocks and generation of block
+   proofs.
+
+This  `TSS-Ledger-ID` proposal covers changes to the following elements:
+
+* The process of generating the ledger key pair for new networks and existing networks.
+* The process for signing a message with the ledger private key.
+* The process of transferring the ability of the ledger to sign a message from one set of consensus nodes to another.
+* The new state data structures needed to store TSS Key material and the ledger id.
+* The new components needed in the framework to support creating and transferring the ledger key.
+* The modified startup process to initialize the signing capability.
+
+### Goals
+
+The following capabilities are the goals of this proposal:
+
+1. `Ledger Signing API` - The app gives a message to the platform to sign. The platform signs the message with its
+   shares, gossips the signatures, and aggregates share signatures into a ledger signature. The ledger signature is
+   returned to the app through the `PlatformPublisher`.
+2. `TSS Bootstrap on Existing Network` - Able to setup an existing network with a ledger public/private key pair.
+3. `TSS Bootstrap for New Network` - Able to setup a new network with a ledger public/private key pair.
+4. `Keying The Next Roster` - Able to transfer the ability to sign a message with the ledger private key from one set of
+   consensus nodes to another.
+
+### Non-Goals
+
+The following are not goals of this proposal:
+
+1. Achieving a fully dynamic address book life-cycle.
+2. Block Proofs or the signing of blocks.
+3. Verification of signed messages beyond the consensus node.
+    * It is not within the scope of this proposal and effort to develop, test, and deploy a smart contract that can
+      verify a block proof. It is expected that mathematics works and that our use of the ALT-BN128 curve will
+      dovetail with the
+      [precompiles that are available in the EVM](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1829.md).
```

Comment:
I agree that this is not in the scope of work to be performed by the platform team, but it would be a very good idea to see if someone from the SC team could do this. We should have some idea of the gas required for verifying the aggregate signature and double check that it does in fact work. Would you be willing to include it in this proposal as long as the work is done by someone on the SC team?

Reply:
I've modified the goals as follows: 
 * removal of the ledger signing API 
 * added testing of the signature scheme with verification by a smart contract.